### PR TITLE
implem exo vm stop

### PR DESCRIPTION
**Pull request by pierre-emmanuelJ**
Signed-off-by: Pierre-Emmanuel Jacquier <pierre-emmanuel.jacquier@epitech.eu>

### CODE REVIEW

File: cmd/exo/cmd/vm.go [hunk 1]
```diff
@@ -11,11 +11,23 @@ var vmCmd = &cobra.Command{
...
 }
 
-func getVMWithNameOrID(cs *egoscale.Client, name string) (*egoscale.VirtualMachine, error) {
+// nolint
+const (
+	starting = "Starting"
+	running  = "Running"
+	stoping  = "Stopping"
+	stopped  = "Stopped"
+)
```
Comment (greut) [inline cmd/exo/cmd/vm.go]: Is it worth creating it in egoscale then?

https://github.com/apache/cloudstack/blob/893a88d225276e45f12f9490e6af2c94a81c2965/api/src/main/java/com/cloud/vm/VirtualMachine.java#L46-L55

Comment (pierre-emmanuelJ) [reply cmd/exo/cmd/vm.go]: I put it in egoscale is better

File: virtual_machines.go [hunk 2]
```diff
@@ -11,6 +11,20 @@ import (
 	"net/url"
 )
 
+// nolint
```
Comment (greut) [inline virtual_machines.go]: it's called cheating

File: cmd/exo/cmd/vm_stop.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+		}
+
+		for _, v := range args {
+			if err := asyncStopWithCtx(gContext, v); err != nil {
+				fmt.Fprintln(os.Stderr, err) // nolint: errcheck
+			} else {
+				fmt.Println("\nStopped !")
+			}
+		}
+		return nil
```
Comment (greut) [inline cmd/exo/cmd/vm_stop.go]: I don't like this `return nil`

Comment (pierre-emmanuelJ) [reply cmd/exo/cmd/vm_stop.go]: I dont want to stop all args vm stopping so I just display error on stderr and continue the flow.
that why I return nil at the end

Comment (greut) [reply cmd/exo/cmd/vm_stop.go]: gather all the errors and return one of them in case of a failure.

File: cmd/exo/cmd/vm_stop.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+		if err != nil {
+			errorReq = err
+			return false
+		}
+
+		if jobResult.JobStatus == egoscale.Success {
+			return false
+		}
+		return true
+	})
```
Comment (greut) [inline cmd/exo/cmd/vm_stop.go]: do the final `Println` here.

**Comment by greut** [comment]
```console
% exo vm stop foo bar
could not stop "foo": virtual machine is not running
could not stop "bar": API error ParamError 431 (CSErrorCode(0) 0): not found, query: apikey=EXOc94bc655ff901b7218a5c3cd, command=listVirtualMachines, name=bar, response=json

% exo vm stop foo bar
Stopping "foo" .. success.
could not stop "bar": API error ParamError 431 (CSErrorCode(0) 0): not found, query: apikey=EXOc94bc655ff901b7218a5c3cd, command=listVirtualMachines, name=bar, response=json

% echo $?
1
```

**Comment by pierre-emmanuelJ** [comment]
okay

File: cmd/exo/cmd/vm_stop.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+			return errors.New(b.String())
+		}
+
+		return nil
+	},
+}
+
+// stopVirtualMachine stop a virtual machine instance
+func stopVirtualMachine(vmName string) error {
+	vm, err := getVMWithNameOrID(cs, vmName)
```
Comment (greut) [inline cmd/exo/cmd/vm_stop.go]: `cs` is a global variable. We don't have to repeat it here.

**Pull request merged**